common/reflect_resource.go

```diff
@@ -46,6 +63,35 @@ func reflectKind(k reflect.Kind) string {
 	return n
 }
 
+func chooseFieldNameWithAliases(typeField reflect.StructField, fieldNamePath []string, aliases map[string]string) string {
```

Comment:
nit: what do you think about having one function for choosing the field name that does:
1. Return `getAlias()` if not empty, or else
2. Check for the alias in the map, or else
3. Return getJsonFieldName

That way, in case we need to get the alias elsewhere, we don't need to use the `if` to call the right function.

Reply:
yep, sounds good, let me update that